src/test/ui/self/elision/ref-self-async.stderr

```diff
@@ -4,69 +4,111 @@ error[E0623]: lifetime mismatch
+   |
+LL |     async fn pin_ref_Self(self: Pin<&Self>, f: &u32) -> &u32 {
+   |                                                         ^^^^
 
 error[E0623]: lifetime mismatch
   --> $DIR/ref-self-async.rs:41:9
    |
 LL |     async fn box_box_ref_Self(self: Box<Box<&Self>>, f: &u32) -> &u32 {
    |                                             -----                ----
    |                                             |
-   |                                             this parameter and the return type are declared with different lifetimes...
+   |                                             this parameter and the returned future are declared with different lifetimes...
 LL |         f
-   |         ^ ...but data from `f` is returned here
+   |         ^ ...but data from `f` is held across an await point here
+   |
+note: `async fn`s return an `impl Future<Output={return type}`
+  --> $DIR/ref-self-async.rs:40:66
+   |
+LL |     async fn box_box_ref_Self(self: Box<Box<&Self>>, f: &u32) -> &u32 {
+   |                                                                  ^^^^
 
 error[E0623]: lifetime mismatch
   --> $DIR/ref-self-async.rs:45:9
    |
 LL |     async fn box_pin_ref_Self(self: Box<Pin<&Self>>, f: &u32) -> &u32 {
    |                                             -----                ----
    |                                             |
-   |                                             this parameter and the return type are declared with different lifetimes...
+   |                                             this parameter and the returned future are declared with different lifetimes...
 LL |         f
-   |         ^ ...but data from `f` is returned here
+   |         ^ ...but data from `f` is held across an await point here
+   |
+note: `async fn`s return an `impl Future<Output={return type}`
+  --> $DIR/ref-self-async.rs:44:66
+   |
+LL |     async fn box_pin_ref_Self(self: Box<Pin<&Self>>, f: &u32) -> &u32 {
+   |                                                                  ^^^^
 
 error[E0623]: lifetime mismatch
   --> $DIR/ref-self-async.rs:49:9
    |
 LL |     async fn wrap_ref_Self_Self(self: Wrap<&Self, Self>, f: &u8) -> &u8 {
    |                                            -----                    ---
    |                                            |
-   |                                            this parameter and the return type are declared with different lifetimes...
+   |                                            this parameter and the returned future are declared with different lifetimes...
 LL |         f
-   |         ^ ...but data from `f` is returned here
+   |         ^ ...but data from `f` is held across an await point here
+   |
+note: `async fn`s return an `impl Future<Output={return type}`
+  --> $DIR/ref-self-async.rs:48:69
+   |
+LL |     async fn wrap_ref_Self_Self(self: Wrap<&Self, Self>, f: &u8) -> &u8 {
+   |                                                                     ^^^
```

Comment:
Looking at this seems like we could use a `span_label` instead of a `span_note` for the new message.

```
error[E0623]: lifetime mismatch
  --> $DIR/ref-self-async.rs:29:9
   |
LL |     async fn ref_Self(self: &Self, f: &u32) -> &u32 {
   |                             -----              ---- `async fn`s return an `impl Future<Output = &u32>`
   |                             |
   |                             this parameter and the returned future are declared with different lifetimes...
LL |         f
   |         ^ ...but data from `f` is held across an await point here
```

@tmandry How busy do you think the output looks like?

Reply:
Your example looks good to me!